src/sim/Action.hh

```diff
@@ -0,0 +1,57 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file Action.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "base/Macros.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Broad categories of events applied to a particle in a track.
+ */
+enum class Action
+{
+    failure, //!< Recoverable error in sampling
+    birth,   //!< Primary particle from an event or creation of a secondary
+    cell,    //!< Enter a new cell
```

Comment:
In Geant, we usually refer to Volume(s), cell is the exceptions (there are few implementation of voxelization).

A related enum from GeantV:
```
enum TrackStatus_t { kAlive, kKilled, kInFlight, kBoundary, kExitingSetup, kPhysics, kPostponed, kNew };
```
```suggestion
    onBoundary,    //!< reached the boundary of a volume or cell.
```

Reply:
Yep, this is an artifact of translation from shift... the nomenclature in the `geometry` branch is "volume" so I'll update to that. As far as the tracking/stepping loop goes, I don't think of the surface crossing event as going from "not on a surface" to "on a surface", but rather as a change in state of the volume ID. So is 
```suggestion
    volume,    //!< Enter a new volume
```
acceptable?

Reply:
Well, at least in some process (multiple scattering), there is a different treatment when near the surface (but I guess that more a state ("being on the boundary and/or crossing from one volume to another) than an action.
But reading "Action::volume", I think "being in a volume", (i.e. 'volume' is not an action/verb), so maybe:
```
   boundaryCrossing
```
or 
```
   volumeChange
```

Reply:
Yes, that's a good point -- the verb tenses (or lack of verbs, as you say) are inconsistent in the current enum. I will clean those up for consistency, including renaming the "volume" one to read as a verb, `changed_volume`.